Failuer to parse `slog.Level` with `flag redefined`

Attempting to fill fields from a structure containing `slog.Level`, which is a redefinition of `int`, panics with `flag: redefined: <flag-name>` at runtime:

```
/tmp/go-build2849440774/b001/exe/test flag redefined: level
panic: /tmp/go-build2849440774/b001/exe/test flag redefined: level

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc0000bc000, {0x53f268, 0xc0000900a0}, {0xc0000141e0, 0x5}, {0x0, 0x0})
        /nix/store/j9hk9x78z301jqldfyvw72z24vb992qi-go-1.22.6/share/go/src/flag/flag.go:1028 +0x37d
flag.(*FlagSet).Func(...)
        /nix/store/j9hk9x78z301jqldfyvw72z24vb992qi-go-1.22.6/share/go/src/flag/flag.go:980
github.com/itzg/go-flagsfiller.(*FlagSetFiller).processCustom(0xc0000141e0?, {0x504a20, 0xc0000141d0}, 0x51bca8, 0x0?, {0x0?, 0xc000014210?}, 0xc0000bc000, {0xc0000141e0, 0x5}, ...)
        /home/hunter/go/pkg/mod/github.com/itzg/go-flagsfiller@v1.14.0/flagset.go:621 +0x292
github.com/itzg/go-flagsfiller.(*FlagSetFiller).processInt(0x0?, {0x504a20?, 0xc0000141d0?}, 0xfe?, {0x0?, 0xc0000141d0?}, 0x20a20?, {0xc0000141e0?, 0x5e8420?}, {0x0, ...}, ...)
        /home/hunter/go/pkg/mod/github.com/itzg/go-flagsfiller@v1.14.0/flagset.go:410 +0x1cd
github.com/itzg/go-flagsfiller.(*FlagSetFiller).processField(0xc000074040, 0xc0000bc000, {0x504a20, 0xc0000141d0}, {0x4e46fb, 0x5}, {0x540c60, 0x4faf60}, {0x0, 0x0})
        /home/hunter/go/pkg/mod/github.com/itzg/go-flagsfiller@v1.14.0/flagset.go:224 +0x759
github.com/itzg/go-flagsfiller.(*FlagSetFiller).walkFields.func1({{0x4e46fb, 0x5}, {0x0, 0x0}, {0x540c60, 0x4faf60}, {0x0, 0x0}, 0x0, {0xc0000141d8, ...}, ...}, ...)
        /home/hunter/go/pkg/mod/github.com/itzg/go-flagsfiller@v1.14.0/flagset.go:98 +0x1b3
github.com/itzg/go-flagsfiller.(*FlagSetFiller).walkFields(0xc000074040, 0xc0000bc000, {0x0?, 0x7f23091f3108?}, {0x4f8b20?, 0xc0000141d0?, 0xc000074040?}, {0x540c60, 0x4f8b20})
        /home/hunter/go/pkg/mod/github.com/itzg/go-flagsfiller@v1.14.0/flagset.go:156 +0x6fc
github.com/itzg/go-flagsfiller.(*FlagSetFiller).Fill(0xc000074040, 0xc0000bc000, {0x4ee0a0?, 0xc0000141d0?})
        /home/hunter/go/pkg/mod/github.com/itzg/go-flagsfiller@v1.14.0/flagset.go:54 +0x1de
main.main()
        /tmp/tmp.dPTGjwn3vd/main.go:16 +0x47
exit status 2
```

Minimal example:

```go
package main

import (
	"flag"
	"log/slog"

	"github.com/itzg/go-flagsfiller"
)

func main() {
	var args struct {
		Level slog.Level
	}
	flagsfiller.New().Fill(flag.CommandLine, &args)
}
```

I'm new to go so sorry in advance if i missed some debug info.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -17,7 +17,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: '1.19'
+        go-version-file: go.mod
 
     - name: Test
       run: go test -v ./...

--- a/additional.go
+++ b/additional.go
@@ -1,0 +1,19 @@
+package flagsfiller
+
+import (
+	"log/slog"
+	"reflect"
+)
+
+func init() {
+	RegisterSimpleType(slogLevelConverter)
+}
+
+func slogLevelConverter(s string, _ reflect.StructTag) (slog.Level, error) {
+	var level slog.Level
+	err := level.UnmarshalText([]byte(s))
+	if err != nil {
+		return slog.LevelInfo, err
+	}
+	return level, nil
+}

--- a/addtional_test.go
+++ b/addtional_test.go
@@ -2,6 +2,7 @@ package flagsfiller_test
 
 import (
 	"flag"
+	"log/slog"
 	"net"
 	"net/netip"
 	"testing"
@@ -116,4 +117,60 @@ func TestTextUnmarshalerType(t *testing.T) {
 	require.NoError(t, err)
 
 	assert.Equal(t, netip.AddrFrom4([4]byte{1, 2, 3, 4}), config.Addr)
+}
+
+func TestSlogLevels(t *testing.T) {
+	tests := []struct {
+		name     string
+		value    string
+		expected slog.Level
+	}{
+		{
+			name:     "info",
+			value:    "info",
+			expected: slog.LevelInfo,
+		},
+		{
+			name:     "error",
+			value:    "error",
+			expected: slog.LevelError,
+		},
+		{
+			name: "numeric offset",
+			// Borrowed from https://pkg.go.dev/log/slog#Level.UnmarshalText
+			value:    "Error-8",
+			expected: slog.LevelInfo,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var args struct {
+				Level slog.Level
+			}
+
+			var flagset flag.FlagSet
+			err := flagsfiller.New().Fill(&flagset, &args)
+			require.NoError(t, err)
+
+			err = flagset.Parse([]string{"--level", test.value})
+			require.NoError(t, err)
+
+			assert.Equal(t, test.expected, args.Level)
+		})
+	}
+}
+
+func TestSlogLevelWithDefault(t *testing.T) {
+	var args struct {
+		Level slog.Level `default:"info"`
+	}
+
+	var flagset flag.FlagSet
+	err := flagsfiller.New().Fill(&flagset, &args)
+	require.NoError(t, err)
+
+	err = flagset.Parse([]string{})
+	require.NoError(t, err)
+
+	assert.Equal(t, slog.LevelInfo, args.Level)
 }

--- a/docs.go
+++ b/docs.go
@@ -146,6 +146,7 @@ FlagSetFiller also supports following field types:
 - net.IPNet: format used by net.ParseCIDR()
 - net.HardwareAddr (MAC addr): format used by net.ParseMAC()
 - time.Time: format is the layout string used by time.Parse(), default layout is time.DateTime, could be overriden by field tag "layout"
+- slog.Level: parsed as specified by https://pkg.go.dev/log/slog#Level.UnmarshalText, such as "info"
 
 # Environment variable mapping
 

--- a/flagset.go
+++ b/flagset.go
@@ -200,7 +200,7 @@ func (f *FlagSetFiller) processField(flagSet *flag.FlagSet, fieldRef interface{}
 	if isSupportedStruct(fieldRef) {
 		handler := extendedTypes[getTypeName(t)]
 		err = handler(tag, fieldRef, hasDefaultTag, tagDefault, flagSet, renamed, usage, aliases)
-
+		return err
 	}
 
 	switch {

--- a/general.go
+++ b/general.go
@@ -11,7 +11,7 @@ import (
 	"strings"
 )
 
-// this is a list of addtional supported types(include struct), like time.Time, that walkFields() won't walk into,
+// this is a list of additional supported types(include struct), like time.Time, that walkFields() won't walk into,
 // the key is the is string returned by the getTypeName(<type>),
 // each supported type need to be added in this map in init()
 var extendedTypes = make(map[string]handlerFunc)

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/itzg/go-flagsfiller
 
-go 1.19
+go 1.21
 
 require (
 	github.com/iancoleman/strcase v0.3.0

